integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/IntegrajonspunktReceiveImpl.java

```diff
@@ -117,37 +119,58 @@ public CorrelationInformation forwardToNoarkSystem(StandardBusinessDocument inpu
             throw e;
         }
 
-        if (document.isReciept()) {
+        if (document.isReceipt()) {
             Audit.info("Messagetype Receipt", markerFrom(document));
             return new CorrelationInformation();
         }
 
-        // TODO: remove? or move to send leveringskvittering i internal queue?
         Payload payload = document.getPayload();
         byte[] decryptedAsicPackage = decrypt(payload);
         EDUCore eduDocument;
-        try {
-            eduDocument = convertAsicEntrytoEduDocument(decryptedAsicPackage);
-            if (PayloadUtil.isAppReceipt(eduDocument.getPayload())) {
-                Audit.info("AppReceipt extracted", markerFrom(document));
-                registerReceipt(eduDocument);
-                if (!properties.getFeature().isForwardReceivedAppReceipts()) {
-                    Audit.info("AppReceipt forwarding disabled - will not deliver to archive");
-                    return new CorrelationInformation();
+        if (document.isNextMove()) {
+            eduDocument = convertConversationToEducore(document);
+        } else {
+            try {
+                eduDocument = convertAsicEntrytoEduDocument(decryptedAsicPackage);
+                if (PayloadUtil.isAppReceipt(eduDocument.getPayload())) {
+                    Audit.info("AppReceipt extracted", markerFrom(document));
+                    registerReceipt(eduDocument);
+                    if (!properties.getFeature().isForwardReceivedAppReceipts()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS134)